Ally_Desktop_Assistant.py

```diff
@@ -0,0 +1,464 @@
+
+
+def show(data):
+    time_to_show = time.strftime("(%I:%M %p)")
+    chat_history.configure(state=NORMAL)
+    chat_history.insert(END, f"{time_to_show} {data}\n")
+    chat_history.yview_pickplace(END)
+    chat_history.update()
+    chat_history.configure(state=DISABLED)
+    with open(history_filename, "a") as f_obj:
+        f_obj.write(f"{time.ctime()}  {data}\n")
+
+
+def time_am_pm():
+
+    pr_time = time.strftime('%I:%M %p')
+    return pr_time
+
+
+def get_date():
+    mon_dict = {1: 'January', 2: 'February', 3: 'March', 4: 'April', 5: 'May', 6: 'June', 7: 'July', 8: 'August',
+                9: 'September', 10: 'October', 11: 'November', 12: 'December'}
+    date_day = str(int(time.strftime("%d")))
+    date_mon_num = int(time.strftime("%m"))
+    date_mon = mon_dict[date_mon_num]
+    date_year = str(int(time.strftime("%Y")))
+    date_week_day = (time.strftime("%A"))
+    today_date = f"{date_week_day} the {date_day}th of {date_mon} {date_year}"
+    return today_date
+
+
+def show_status(string):
+    status_var.set(string)
+    status_label.update()
+
+
+def change_voice_code(event):
+    global temp_voice_code
+
+    voice_type = (voice_var.get())
+    filename = data_filename
+    try:
+        with open(filename, "r") as f_obj:
+            data_dict = json.load(f_obj)
+            data_dict["voice"] = voice_type
+
+        with open(filename, 'w') as file_object:
+            json.dump(data_dict, file_object)
+
+    except:
+        with open(filename, "w") as file_obj:
+            data_dict_new = {"voice": voice_type}
+            json.dump(data_dict_new, file_obj)
+    temp_voice_code = get_voice_code()
+    tmsg.showinfo("Voice Changed", f"Voice is changed to {voice_type}.")
+
+
+def get_voice_code():
+    try:
+        with open(data_filename) as file_obj:
+            voice_type = json.load(file_obj)["voice"]
+            voice_type = voice_type.lower()
+            if voice_type == "male":
+                voice_code = 0
+            elif voice_type == "female":
+                voice_code = 1
+            else:
+                voice_code = 1
+    except:
+        voice_code = 1
+
+    return voice_code
+
+
+def speak(string):
+    """ This functions accepts a string and then speaks it. """
+    # 1- female
+    # 0 - male
+    show_status("Busy")
+    show(f"{name.title()} : {string}\n")
+    engine = pyttsx3.init('sapi5')
+    type_voices = engine.getProperty('voices')
+    engine.setProperty('voice', type_voices[temp_voice_code].id)
+    engine.say(string)
+    engine.runAndWait()
+    show_status("Ready now")
+
+
+def take_command():
+    r = sr.Recognizer()
+    with sr.Microphone() as source:
+        show_status("Listening...")
+        r.pause_threshold = 1
+        audio = r.listen(source)
+        try:
+            show_status("Recognizing")
+            query = r.recognize_google(audio, language='en-in')
+            return_val = query
+
+        except Exception as e:
```

Comment:
Don't do the **Catch 'Em All**. Do the proper exception handling. Please check 
https://pythonspot.com/speech-recognition-using-google-speech-api/ and https://www.programcreek.com/python/example/107720/speech_recognition.RequestError to get an insight of exception handling for google recognize